src/Psalm/Internal/Type/Comparator/GenericTypeComparator.php

```diff
@@ -136,36 +136,33 @@ public static function isContainedBy(
                 && !$container_type_part instanceof TIterable
                 && !$container_param->hasTemplate()
                 && !$input_param->hasTemplate()
+                && !($container_type_params_covariant[$i] ?? false)
             ) {
-                if ($input_param->containsAnyLiteral()) {
+                if ($input_param->containsAnyLiteral() || $input_param->containsAnyNonEmptiness()) {
```

Comment:
In the past, I wanted to remove it, but changed my mind. Without this, it can be really hard to write tests or fixtures where the values are hardcoded.

Moreover, I have added here `Box<non-empty-list<T>>` to `Box<list<T>>` specialization.